### PR TITLE
[N3] Make format types stricter using tempate literals

### PR DESCRIPTION
**Pull request by jeswr**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/rdfjs/N3.js/issues/255>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by rubensworks — APPROVED** [review]
LGTM, but might be good if @RubenVerborgh double-checks this one.

File: types/n3/index.d.ts [hunk 1]
```diff
@@ -149,8 +150,35 @@ export class Lexer {
...
+  // Multipart types
+  | 'message' | 'multipart';
+
+export type BaseFormat =
+  | 'turtle'
+  | 'triG'
+  | 'N-Triples'
+  | 'N-Quads'
+  | 'n3'
+  | 'Notation3';
```
Comment (RubenVerborgh) [inline types/n3/index.d.ts:167]: Shall we perhaps keep casing consistent here? Should be `TriG` and `Turtle` and `N3` for me.

**Review by RubenVerborgh — APPROVED** [review]

File: types/n3/index.d.ts [hunk 2]
```diff
@@ -149,8 +150,35 @@ export class Lexer {
...
+
+export type BaseFormatVariant =
+  | BaseFormat
+  | Lowercase<BaseFormat>
+  | Capitalize<BaseFormat>;
+
+export type Star = '*' | 'star';
```
Comment (RubenVerborgh) [inline types/n3/index.d.ts:174]: Maybe allow `-star` too? So we can `Turtle-star` as in the spec?

**Review by rubensworks — APPROVED** [review]

**Comment by jeswr** [comment]
Ready to merge

**Comment by TallTed** [comment]
This may have been merged too quickly...

Note that the RDF-star CG has settled on `-star` as the suffix for both the data model and the adjusted serializations (e.g., RDF-star, Turtle-star) -- which include some but perhaps not all formats your code expands to -- all of which is still in a draft-CG-report, so support for these serializations may be premature -- and your "stricter" types do not in fact appear to be "stricter" (as `Turtle*`, `Turtlestar`, and `Turtle-star` make one type look like three, to me).

**Comment by RubenVerborgh** [comment]
As far as I can see, all works as expected, which is:
- full MIME types are accepted
- shorthands such as "Turtle" are accepted

**Comment by RubenVerborgh** [comment]
This change is not backwards-compatible and breaks clients passing strings. I suggest to revert, or at least add `string | ` back in.

**Pull request merged**